### PR TITLE
Adds Moffuchi's Pizzeria ruin

### PR DESCRIPTION
**Pull request by Capsandi**
## About The Pull Request

This PR adds a new ruin to the surface of Icebox; Moffuchi's Pizza Parlor, ran by moth chef Moffuchi and his apprentice, who start the shift locked in their own freezers. They are both ghostroles tasked with improving the restaurant and selling pizza. They are given a kitchen and tools to rearrange the layout of the restaurant while they wait for miners or bored assistants to bump into them. They are not given mining implements or any coats to survive in the caves with. The restaurant's starting layout can be seen below.
mothpizza
And heres an up to date picture of the ruin in Strongdmm
image

## Why It's Good For The Game

New ruins for icebox helps make the moon feel less like a lavaland reskin and more like its own place. 
The pizzeria has many possible interactions with the crew, from taking payment in delicious clothing to making deliveries if supplied with coats and a way to the station. You could even celebrate Ian's birthday at Moffuchi's.

## Changelog
:cl:
add: Added the Moffuchi's Pizzaria Ruin to the Icemoon Caves and two associated ghost roles
/:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
is this a fnaf reference

**Comment by necromanceranne** [comment]
Where is the parking meter. Is it possible for this to spawn on the station z-level?

**Comment by Capsandi** [comment]
It Is currently not possible for it to spawn on the station Z, only the lower Z levels. I didn't want it to spawn right next to an airlock into the station or something silly like that

**Comment by Doosteroni** [comment]
Give them a mop and bucket

**Comment by Doosteroni** [comment]
> It Is currently not possible for it to spawn on the station Z, only the lower Z levels. I didn't want it to spawn right next to an airlock into the station or something silly like that

Might want to lock it to the upper end of the ice caves so literal fucking dragons don’t infinitely spawn out of a portal next to it

**Comment by Capsandi** [comment]
They have the shutters, which in my tests deterred gaining aggro from anything spawning nearby. And portals very rarely spawned close as possible to the building. Also the mop and bucket is there in the hands of the waiter moth

**Comment by Whoneedspacee** [comment]
i love it

**Comment by Kaffe-work** [comment]
why not have it spawn on the upper level, really? is there a problem if it is close by to the station, since it's interesction/RP relies upon the station? the only issue is someone ided time for vengeance, but that should just be taken care of by a ghost role-ban

**Comment by Capsandi** [comment]
Good point, I'll move it to the top level

File: code/datums/ruins/icemoon.dm [hunk 1]
```diff
@@ -38,6 +38,12 @@
 	prefix = "_maps/RandomRuins/AnywhereRuins/"
 	suffix = "fountain_hall.dmm"
 
+/datum/map_template/ruin/icemoon/surface/moth_pizza
+	name = "Moffuchis Pizzaria"
```
Comment (Fikou) [inline code/datums/ruins/icemoon.dm:42]:
```suggestion
	name = "Moffuchi's Pizzeria"
```

File: code/datums/ruins/icemoon.dm [hunk 2]
```diff
@@ -38,6 +38,12 @@
...
+	id = "moth-pizzaria"
```
Comment (Fikou) [inline code/datums/ruins/icemoon.dm:43]:
```suggestion
	id = "moth-pizzeria"
```

File: code/game/area/areas/ruins/icemoon.dm [hunk 3]
```diff
@@ -9,3 +9,11 @@
 	icon_state = "dk_yellow"
 	mood_bonus = 10
 	mood_message = "<span class='nicegreen'>This place reminds me of home.</span>\n"
+
+/area/ruin/unpowered/mothresturant
+	name = "Moffuchis Pizza"
```
Comment (Fikou) [inline code/game/area/areas/ruins/icemoon.dm:14]:
```suggestion
	name = "Moffuchi's Pizzeria"
```

File: code/game/area/areas/ruins/icemoon.dm [hunk 4]
```diff
@@ -9,3 +9,11 @@
...
+	icon_state = "dk_yellow"
+
+/area/ruin/unpowered/mothkitchen
+	name = "Moffuchis Kitchen"
```
Comment (Fikou) [inline code/game/area/areas/ruins/icemoon.dm:18]:
```suggestion
	name = "Moffuchi's Kitchen"
```

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 5]
```diff
@@ -314,6 +314,71 @@
 	l_pocket = /obj/item/tank/internals/emergency_oxygen
 	r_pocket = /obj/item/flashlight/glowstick
 
+//Moffuchi's pizza parlor,located in the icemoon's caves, is home to chef Moffuchi and his apprentice. They run their resturant for hungery miners and adventurers, but mostly for cash
+/obj/effect/mob_spawn/human/mothchef
+	name = "locked freezer"
+	desc = "You can hear fluttering from inside this freezer,<i>is there a moth in there</i>?"
+	mob_name = "pizzaria manager"
```
Comment (Fikou) [inline code/game/objects/structures/ghost_role_spawners.dm:321]:
```suggestion
	mob_name = "pizzeria manager"
```

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 6]
```diff
@@ -314,6 +314,71 @@
...
+	icon = 'icons/obj/crates.dmi'
+	icon_state = "freezer"
+	roundstart = FALSE
+	death = FALSE
+	random = TRUE
+	id_job = "Pizzaria Manager"
```
Comment (Fikou) [inline code/game/objects/structures/ghost_role_spawners.dm:327]:
```suggestion
	id_job = "Pizzeria Manager"
```

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 7]
```diff
@@ -314,6 +314,71 @@
...
+	var/mob/living/carbon/human/H = L
+	var/obj/item/worn = H.wear_id
+	var/obj/item/card/id/id = worn.GetID()
+	id.registered_name = L.real_name
+	id.update_label()
+
+/obj/effect/mob_spawn/human/mothsouschef
+	name = "locked freezer"
+	desc = "You can hear fluttering from inside this freezer,<i>is there a moth in there</i>?"
+	mob_name = "pizzaria assistant manager"
```
Comment (Fikou) [inline code/game/objects/structures/ghost_role_spawners.dm:359]:
```suggestion
	mob_name = "pizzeria assistant manager"
```

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 8]
```diff
@@ -314,6 +314,71 @@
...
+	icon = 'icons/obj/crates.dmi'
+	icon_state = "freezer"
+	roundstart = FALSE
+	death = FALSE
+	random = TRUE
+	id_job = "Pizzaria Assistant Manager"
```
Comment (Fikou) [inline code/game/objects/structures/ghost_role_spawners.dm:365]:
```suggestion
	id_job = "Pizzeria Assistant Manager"
```

**Review by Fikou — CHANGES_REQUESTED** [review]

**Comment by Fikou** [comment]
what the f*ck is a pizzaria....

File: code/game/objects/structures/ghost_role_spawners.dm (hunk 8 above)
Comment (Capsandi) [reply code/game/objects/structures/ghost_role_spawners.dm:365]: I remember finding out i was spelling it wrong early on, then I told myself I would find and replace them with the correct spelling. I guess i forgot to do that. Thanks for catching that.

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
policy support (`assignedrole` on the spawner)

**Comment by ExcessiveUseOfCobblestone** [comment]
how do they get more pizza?

**Comment by Fikou** [comment]
you probably should add a cow, some tomato and wheat seeds and a tray

**Comment by Iatots** [comment]
Could've also just given the windows a paint layer to make them non transparent.

**Comment by Capsandi** [comment]
Moffuchi starts out with wheat seeds, and his apprentice starts out with tomato seeds. Theres sticky tape in the trashcan to make an improvised pickaxe for sandstone if you somehow sell all your pizza without asking for anything of use in return. I just added a mouse to one of the back rooms you can feed cheese to if you need meat. Custom pizzas only need wheat for the dough, and improving your selection of ingredients further may require you to barter for it while selling the pizza. 
I could instead give them a supply pod console which only allows the purchase of cooking supplies and targets the restaurant instead of the cargo bay.

**Comment by ATH1909** [comment]
Could you move the important stuff out of the inventories of the moths and into the ruin itself, so Moffuchi isn't screwed over if no ghosts decide to possess his apprentice?

Also, could you add, like, a teleporter beacon or a GPS signal to this ruin or something, so people can actually find it?

Do the moths have radios?

**Comment by ATH1909** [comment]
Uh, how are the moths supposed to get to the trash bin outside that contains the sticky tape if it's really cold outside and there's no path from the trash bin back into the pizza parlor that doesn't require mining or destroying stuff?

**Comment by ATH1909** [comment]
Do they have any way to recharge their generator, other than mining ore in the cold?

**Comment by ATH1909** [comment]
Do they have any way to breathe outside while they're mining sand for plant plots or grabbing the sticky tape?

**Comment by Capsandi** [comment]
It's in the trash bin inside, impossible to miss and i will be putting an updated screenshot of the map editor up, the cells last for 45 minutes before turning off the lighting, longer than most rounds outside of manuel, and i may close this pr and work on the snowflakey pod console because not every restaurant has a cow lmao

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 9]
```diff
@@ -314,6 +314,73 @@
...
+	//Possible names based on famous chefs, real or fictional.
+	var/moth_name = pick("Hank","Gordon","Guy","Chef","Willy","Brad","Bobby","Alton")
+	L.fully_replace_character_name(null,"[moth_name] Moffuchi")
+	var/mob/living/carbon/human/H = L
+	var/obj/item/worn = H.wear_id
+	var/obj/item/card/id/id = worn.GetID()
+	id.registered_name = L.real_name
+	id.update_label()
+
+/obj/effect/mob_spawn/human/mothsouschef
```
Comment (ATH1909) [inline code/game/objects/structures/ghost_role_spawners.dm:357]: why not make this a subtype of the other spawner?

Comment (Capsandi) [reply code/game/objects/structures/ghost_role_spawners.dm:357]: That's exactly what i was thinking, but i didn't think about doing it the other way around, I'll get to that in a moment

**Comment by Capsandi** [comment]
Closing due to an idea i had which would cut down on the amount of gear the moths require to be self sufficient. This will take a while to code so I'm closing this for now. The feedback was very helpful!

**Comment by Pumpkinoe** [comment]
Also make sure there's a redtext telling them that abandoning their resteraunt completely would be a detriment to buisness or something

**Comment by Nikitaw99** [comment]
Buttons inside the bathroom stalls that bolt the bathroom stall door would be neat

**Pull request closed without merging**